src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -0,0 +1,47 @@
+package org.broadinstitute.dsde.rawls.dataaccess.slick
+
+import java.sql.Timestamp
+import java.util.UUID
+
+import org.joda.time.DateTime
+
+/**
+ * Created by dvoet on 2/4/16.
+ */
+case class WorkspaceRecord(
+  namespace: String,
+  name: String,
+  id: UUID,
+  bucketName: String,
+  createdDate: Timestamp,
+  lastModified: Timestamp,
+  createdBy: String,
+  isLocked: Boolean,
+  workspaceEntityId: Option[Long]
+)
+
+// TODO access control
+
+trait WorkspaceComponent {
+  this: DriverComponent with EntityComponent =>
+
+  import driver.api._
+
+  class WorkspaceTable(tag: Tag) extends Table[WorkspaceRecord](tag, "WORKSPACE") {
+    def namespace = column[String]("namespace")
+    def name = column[String]("name")
+    def id = column[UUID]("id", O.PrimaryKey)
+    def bucketName = column[String]("bucket_name")
+    def createdDate = column[Timestamp]("created_date")
+    def lastModified = column[Timestamp]("last_modified")
+    def createdBy = column[String]("created_by")
+    def isLocked = column[Boolean]("is_locked")
+    def workspaceEntityId = column[Option[Long]]("workspace_entity_id")
+
+    def workspaceEntity = foreignKey("FK_WORKSPACE_ENTITY", workspaceEntityId, entityQuery)(_.id.?)
```

Comment:
an alternative would be to have an attribute table without entity id then 2 join tables, one for entity and one for workspace. Firehose used to do it this way and currently does it the way this PR does.